specs/datapackages/README.md

```diff
@@ -0,0 +1,130 @@
+
+2. Allow exporting CKAN datasets as Data Packages, regardless of whether they were imported from Data Packages in a first instance or not.
+
+3. Support schema definition for CKAN resources, allowing publishers to describe the contents of published files
+
+
+For achieving this it seems sensible to reuse as much stuff as we can from ckanext-datapackager, extending it if necessary
+
+The first two points could be implemented either in core or separate extension, but I'd like to propose that the third regarding resource schemas is implemented in core (in a generic form). It seems central enough to all future work around data import, cleaning, etc (eg closer integration with DataStore) to justify it.
+
+## Field mapping    
+
+Both for transforming Data Packages into CKAN datasets and to generate Data Packages from CKAN datasets we need to have a good mapping between the two.
+
+CKAN **dataset** core fields have essentially the same equivalents on Data Packages (`title`, `description` , `keywords`, ...). Potential issues:
+
+* `name` - collision with existing CKAN one (we can just add an incremental number)
+* `license` - mapping to the ones available on the CKAN instance. **Q:** What happens if we can not map it?
+* `author` / `contributors` - these refer to the datapackage creators not the data ones (`sources` is used for this) **Q:** Should be bother trying to map to CKAN's `author` / `author_email` or `maintainer` /  `maintainer_email`?
+
+Data Packages can have extra fields not present in the CKAN schema:
+* `sources` (name, web, email)
+* `image`
+* `base`
+* `dataDependencies` (ordering of the hash keys is important)
+* `schemas` 
+* Arbitrary fields added to the Data Package
+
+If we don't store the original Data Package somewhere we'll need to store these as extras. 
+
+As per **resources**,  Data Packages `resources` can point the actual data files using `url` or `path` + `base`, or have the data embedded with a `data` field. (The current spec allows all of them to be present at the same time, which hopefully we can restrict https://github.com/dataprotocols/dataprotocols/issues/223). If Data Packages have `data` embedded we can push it to the DataStore to make it available.
+
+Most fields map to CKAN core ones. The ones that are not defined are:
+
+* `encoding`
+* `hash`
+* `sources`
+* `license`
+
+This is not as problematic as dataset fields, as we can add arbitrary top level fields to resources.
+
+## Importing Data Packages
+
+### Implementation
+
+At a higher level this seems quite straight-forward: a  `datapackage_import` or similar action which inherits the auth from `package_create` and `package_update` (eg you need to belong to an org by default etc). The action will validate the JSON object, transform it to a CKAN dict and create or update the relevant package. A controller will handle requests from the UI (at `/datapackage/import`).
```

Comment:
I guess the principle here is that the metadata should be kept close to the data. So if a hacker or organization publishes both then you should push both or pull both. I can see there's also the case where a hacker creates metadata that points to an official source, so it's not possible to require they come via the same method. So you're write, for the case of data sources like hackers or organizations that choose to make the CKAN their primary place of distributing data, allowing upload of a data package makes some sense.

And yes I see from other bits of your doc that you intended `datapackage_import` to do both create and update, as appropriate, so that's fine.